crates/api/src/local_user/save_settings.rs

```diff
@@ -41,8 +41,11 @@ pub async fn save_user_settings(
 
   if let Some(Some(email)) = &email {
     let previous_email = local_user_view.local_user.email.clone().unwrap_or_default();
-    // Only send the verification email if there was an email change
-    if previous_email.ne(email) {
+    // if email was changed, check that it is not taken and send verification mail
+    if &previous_email != email {
+      if LocalUser::is_email_taken(&mut context.pool(), email).await? {
+        return Err(LemmyErrorType::EmailAlreadyExists)?;
```

Comment:
In general it's not good to tell a user this information because it allows email enumeration attacks (finding out which users have an account in the system which then allows easier brue-forcing of accounts based on public password leak lists). The signup flow also has a leak but that might have more barriers (e.g. captcha) so I'm still mentioning it as good practice.

more info on what better responses would be e.g. here https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection

Reply:
The "clean" way I guess would be to just return "ok" always with a message of "an email has been sent to the new email to verify it) and to send two different emails, one saying "could not change email since another account already exists" and one saying "click here to verify your email".

I guess that might be too much effort for this change though. Email enumeration *will* happen though when lemmy gets popular enough and cause account takeovers for people that like to reuse passwords.

Reply:
Makes sense, but that will require updating translation so Id rather do it in another PR. Opened https://github.com/LemmyNet/lemmy/issues/4021